models/glif_psc.cpp

```diff
@@ -0,0 +1,578 @@
+  , threshold_( -51.68 - p.E_L_ ) // in mV
+  , ASCurrents_( p.asc_init_ )    // in pA
+  , I_( 0.0 )                     // in pA
+{
+  y1_.clear();
+  y2_.clear();
+}
+
+/* ----------------------------------------------------------------
+ * Parameter and state extractions and manipulation functions
+ * ---------------------------------------------------------------- */
+
+void
+nest::glif_psc::Parameters_::get( DictionaryDatum& d ) const
+{
+  def< double >( d, names::V_th, th_inf_ + E_L_ );
+  def< double >( d, names::g, G_ );
+  def< double >( d, names::E_L, E_L_ );
+  def< double >( d, names::C_m, C_m_ );
+  def< double >( d, names::t_ref, t_ref_ );
+  def< double >( d, names::V_reset, V_reset_ + E_L_ );
+
+  def< double >( d, names::a_spike, a_spike_ );
+  def< double >( d, names::b_spike, b_spike_ );
+  def< double >( d, names::a_reset, voltage_reset_a_ );
+  def< double >( d, names::b_reset, voltage_reset_b_ );
+
+  def< double >( d, names::a_voltage, a_voltage_ );
+  def< double >( d, names::b_voltage, b_voltage_ );
+
+  def< std::vector< double > >( d, names::asc_init, asc_init_ );
+  def< std::vector< double > >( d, names::k, k_ );
+  def< std::vector< double > >( d, names::asc_amps, asc_amps_ );
+  def< std::vector< double > >( d, names::r, r_ );
+  ArrayDatum tau_syn_ad( tau_syn_ );
+  def< ArrayDatum >( d, names::tau_syn, tau_syn_ad );
+  def< bool >( d, names::has_connections, has_connections_ );
+  def< model_type >( d, names::glif_model, glif_model_ );
+}
+
+double
+nest::glif_psc::Parameters_::set( const DictionaryDatum& d )
+{
+  // if E_L_ is changed, we need to adjust all variables defined relative to
+  // E_L_
+  const double ELold = E_L_;
+  updateValue< double >( d, names::E_L, E_L_ );
+  const double delta_EL = E_L_ - ELold;
+
+  if ( updateValue< double >( d, names::V_reset, V_reset_ ) )
+  {
+    V_reset_ -= E_L_;
+  }
+  else
+  {
+    V_reset_ -= delta_EL;
+  }
+
+  if ( updateValue< double >( d, names::V_th, th_inf_ ) )
+  {
+    th_inf_ -= E_L_;
+  }
+  else
+  {
+    th_inf_ -= delta_EL;
+  }
+
+  updateValue< double >( d, names::g, G_ );
+  updateValue< double >( d, names::C_m, C_m_ );
+  updateValue< double >( d, names::t_ref, t_ref_ );
+
+  updateValue< double >( d, names::a_spike, a_spike_ );
+  updateValue< double >( d, names::b_spike, b_spike_ );
+  updateValue< double >( d, names::a_reset, voltage_reset_a_ );
+  updateValue< double >( d, names::b_reset, voltage_reset_b_ );
+
+  updateValue< double >( d, names::a_voltage, a_voltage_ );
+  updateValue< double >( d, names::b_voltage, b_voltage_ );
+
+  updateValue< std::vector< double > >( d, names::asc_init, asc_init_ );
+  updateValue< std::vector< double > >( d, names::k, k_ );
+  updateValue< std::vector< double > >( d, names::asc_amps, asc_amps_ );
+  updateValue< std::vector< double > >( d, names::r, r_ );
+
+  updateValue< std::vector< double > >( d, names::tau_syn, tau_syn_ );
+  updateValue< model_type >( d, names::glif_model, glif_model_ );
```

Comment:
Should there be any restriction on changing the model type? E.g., you can only change the model type before the simulation starts?